### PR TITLE
Stabilize Deltaspike tests

### PR DESCRIPTION
**Pull request by luvalach**
Problem with stability of Deltaspike tests fixed. Each project will be validated after import.
DeltaspikeTestBase.java formatted.

### CODE REVIEW

File: tests/org.jboss.tools.deltaspike.ui.bot.test/src/org/jboss/tools/deltaspike/ui/bot/test/DeltaspikeTestBase.java [hunk 1]
```diff
@@ -75,45 +79,75 @@ public static void disableSourceLookup(){
...
 			e.printStackTrace();
-			fail("Unable to import "+projectName);
+			fail("Unable to import " + projectName);
 		}
-		
+
 		ProjectExplorer pe = new ProjectExplorer();
 		pe.open();
 		pe.getProject(projectName).select();
 		new ContextMenu("Properties").select();
```
Comment (rawagner) [inline tests/org.jboss.tools.deltaspike.ui.bot.test/src/org/jboss/tools/deltaspike/ui/bot/test/DeltaspikeTestBase.java]: please use org.jboss.reddeer.eclipse.ui.dialogs.ExplorerItemPropertyDialog

File: tests/org.jboss.tools.deltaspike.ui.bot.test/src/org/jboss/tools/deltaspike/ui/bot/test/DeltaspikeTestBase.java [hunk 2]
```diff
@@ -75,45 +79,75 @@ public static void disableSourceLookup(){
...
+		for (TableItem i : new DefaultTable().getItems()) {
+			if (i.getText().equals(sr.getRuntimeNameLabelText(sr.getConfig()))) {
 				i.setChecked(true);
 			} else {
 				i.setChecked(false);
 			}
 		}
+
+		boolean needValidation = false;
+		new DefaultTreeItem("CDI (Contexts and Dependency Injection) Settings").select();
```
Comment (rawagner) [inline tests/org.jboss.tools.deltaspike.ui.bot.test/src/org/jboss/tools/deltaspike/ui/bot/test/DeltaspikeTestBase.java]: and implement & use CDI Settings property page

File: tests/org.jboss.tools.deltaspike.ui.bot.test/src/org/jboss/tools/deltaspike/ui/bot/test/DeltaspikeTestBase.java [hunk 3]
```diff
@@ -75,45 +79,75 @@ public static void disableSourceLookup(){
...
 		}
-		
+
 		ProjectExplorer pe = new ProjectExplorer();
 		pe.open();
 		pe.getProject(projectName).select();
 		new ContextMenu("Properties").select();
-		new DefaultShell("Properties for "+projectName);
+		new DefaultShell("Properties for " + projectName);
 		new DefaultTreeItem("Targeted Runtimes").select();
```
Comment (rawagner) [inline tests/org.jboss.tools.deltaspike.ui.bot.test/src/org/jboss/tools/deltaspike/ui/bot/test/DeltaspikeTestBase.java]: use org.jboss.reddeer.eclipse.wst.common.project.facet.ui.RuntimesPropertyPage

**Comment by luvalach** [comment]
Done, could you check it again please? @rawagner

File: tests/org.jboss.tools.deltaspike.ui.bot.test/src/org/jboss/tools/deltaspike/ui/bot/test/DeltaspikeTestBase.java [hunk 4]
```diff
@@ -75,45 +79,68 @@ public static void disableSourceLookup(){
...
+			log.warn("CDI support is disabled! CDI should be enabled automatically after project import.");
+			log.warn("Enabling CDI support for '%s' project.", projectName);
+			cdiSupportCheckBox.toggle(true);
 		}
-		new PushButton("OK").click();
-		new WaitWhile(new ShellWithTextIsAvailable("Properties for "+projectName));
-		new WaitWhile(new JobIsRunning(),TimePeriod.LONG);
+
+		dialog.ok();
+		new WaitWhile(new ShellWithTextIsAvailable("Properties for " + projectName));
```
Comment (rawagner) [inline tests/org.jboss.tools.deltaspike.ui.bot.test/src/org/jboss/tools/deltaspike/ui/bot/test/DeltaspikeTestBase.java]: you dont need this wait, dialog.ok() contains it

File: tests/org.jboss.tools.deltaspike.ui.bot.test/src/org/jboss/tools/deltaspike/ui/bot/test/DeltaspikeTestBase.java [hunk 5]
```diff
@@ -75,45 +79,68 @@ public static void disableSourceLookup(){
...
+
+		PropertyDialog dialog = new ExplorerItemPropertyDialog(explorerItem);
+		RuntimesPropertyPage propertyPage = new RuntimesPropertyPage();
+		String targetRuntime = sr.getRuntimeNameLabelText(sr.getConfig());
+
+		dialog.open();
+		dialog.select(propertyPage);
+		propertyPage.selectRuntime(targetRuntime);
+
+		new DefaultTreeItem("CDI (Contexts and Dependency Injection) Settings").select();
```
Comment (rawagner) [inline tests/org.jboss.tools.deltaspike.ui.bot.test/src/org/jboss/tools/deltaspike/ui/bot/test/DeltaspikeTestBase.java]: Im still missing "CDI (Contexts and Dependency Injection) Settings" PropertyPage :)

File: tests/org.jboss.tools.deltaspike.ui.bot.test/src/org/jboss/tools/deltaspike/ui/bot/test/exception/DeltaspikeTestInFailureException.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
+package org.jboss.tools.deltaspike.ui.bot.test.exception;
```
Comment (rawagner) [inline tests/org.jboss.tools.deltaspike.ui.bot.test/src/org/jboss/tools/deltaspike/ui/bot/test/exception/DeltaspikeTestInFailureException.java]: copyright

**Comment by luvalach** [comment]
Thanks for feedback @rawagner . Hope everything is fine now.

**Pull request merged**